vllm/v1/core/single_type_kv_cache_manager.py

```diff
@@ -292,15 +280,37 @@ def remove_skipped_blocks(self, request_id: str,
         # No need to remove blocks for full attention.
         pass
 
-    def get_num_common_prefix_blocks(self, request_id: str,
-                                     num_running_requests: int) -> int:
-        blocks = self.req_to_blocks[request_id]
+    def get_num_common_prefix_blocks(
+            self, running_request_id: str, running_request_ids: list[str],
+            transfering_request_ids: list[str]) -> int:
+        """Get common prefix blocks shared by all running and transferring 
+        requests."""
+        if running_request_id not in self.req_to_blocks:
+            return 0
+
+        reference_blocks = self.req_to_blocks[running_request_id]
+        total_requests = len(running_request_ids) + len(
+            transfering_request_ids)
+
+        transferring_blocks = [
+            self.req_to_blocks[req_id] for req_id in transfering_request_ids
+            if req_id in self.req_to_blocks
+        ]
+
         num_common_blocks = 0
-        for block in blocks:
-            if block.ref_cnt == num_running_requests:
-                num_common_blocks += 1
-            else:
+        for i, ref_block in enumerate(reference_blocks):
+
+            if ref_block.ref_cnt < total_requests:
```

Comment:
is this correct? I think if a block is a common block for all running request, but not a common block for transferring request, we should regard it as a common block.

Reply:
yup this actually makes sense!